app/src/main/java/org/oppia/android/app/resumelesson/ResumeLessonActivityPresenter.kt

```diff
@@ -29,28 +50,82 @@ class ResumeLessonActivityPresenter @Inject constructor(
       activity,
       R.layout.resume_lesson_activity
     )
+    this.profileId = profileId
+    this.topicId = topicId
+    this.storyId = storyId
+    this.explorationId = explorationId
+    this.explorationCheckpoint = explorationCheckpoint
+    this.parentScreen = parentScreen
     val resumeLessonToolbar = binding.resumeLessonActivityToolbar
     activity.setSupportActionBar(resumeLessonToolbar)
 
+    retrieveReadingTextSize().observe(
+      activity as ResumeLessonActivity,
+      { result ->
+        (activity as DefaultFontSizeStateListener).onDefaultFontSizeLoaded(result)
+      }
+    )
+
     resumeLessonToolbar.setNavigationOnClickListener {
+      fontScaleConfigurationUtil.adjustFontScale(
+        context = activity,
+        ReadingTextSize.MEDIUM_TEXT_SIZE
+      )
+
```

Comment:
Only activities that have implemented `DefaultFontSizeStateListener` are affected by font size changes. So by default, closing this screen without resetting won't affect the topic screen, making this call redundant.